pykafka/balancedconsumer.py

```diff
@@ -553,6 +566,22 @@ def _topics_changed(self, topics):
         log.debug("Rebalance triggered by topic change")
         self._rebalance()
 
+    def _zookeeper_state_changed(self, state):
+        if state in (KazooState.LOST, KazooState.SUSPENDED):
```

Comment:
There may be a problem if we treat these two states the same way: a suspended connection may still have ownership of its ephemeral nodes. So for the LOST state, what you're doing in L450 above is valid (because the nodes on zookeeper will have gone), but for the SUSPENDED state, doing this might leave the nodes on zookeeper owned by this instance, that is, the instance would then have unilaterally dropped the partitions.